composables/use-issues-by-column.ts

```diff
@@ -0,0 +1,40 @@
+import type { ColumnTitle } from '~/constants/columns'
+import type { IssueStatus } from '~/constants/issues'
+import type { Issue } from '~/types/issues'
+
+function filterIssuesWithoutStatus(issues: Issue[]): Issue[] {
+  const issuesWithoutStatus = issues.filter(
+    (issue) =>
+      !issue.labels.some((label) => label.startsWith(createPartialDataLabel('status'))),
+  )
+
+  return issuesWithoutStatus
+}
+
+function filterIssuesByStatus(issues: Issue[], status: IssueStatus): Issue[] {
+  const issuesWithStatus = issues.filter((issue) =>
+    issue.labels.includes(createDataLabel('status', status)),
+  )
+
+  return issuesWithStatus
+}
+
+function useIssuesByColumn(): Record<ColumnTitle, ComputedRef<Issue[]>> {
+  const route = useRoute('node-rid')
+  const response = useHttpdFetch('/projects/{rid}/issues', {
```

Comment:
Whoah, this function does too much by itself and it's _especially_ problematic because it sneakily contains a pretty expensive network request! Sneakily because it doesn't use the "fetch*" prefix in the name.

The `useRoute` and `useHttpdFetch` have no reason to be so deep in here. They should be high up the dependency tree, perhaps on the page itself (where they are also visible)? 

The pages usually source data and then pass it down the component tree (unless fetching happens in a global store in which case it's still the page, or better yet a router hook for that page's path that triggers the fetching).

We should probably be able to leverage `useAsyncData` and fetch from httpd in there:
- https://nuxt.com/docs/getting-started/data-fetching#useasyncdata
- https://nuxt.com/docs/api/composables/use-async-data

May also be worth checking https://nuxt.com/docs/getting-started/data-fetching#caching-and-refetching

Note that our `useHttpdFetch()` is a wrapper of Nuxt's `useFetch()` and should be able to accept the same config options and be used the same as in the docs.

Reply:
Without the route and fetch, I feel like this would work better as a function that simply transforms the data instead of a `composable`.

I moved `useRoute` and `useHttpdFetch` to the page, created the `groupIssuesByColumn` a function in `utils/issues.ts`, and used it with the `transform` option of `useFetch`. Let me know what you think.

Sidenote: I read about `useAsyncData`, and I'm not sure I understand why we would use it over `useFetch` in this case. From what I understood it just allows you to pass your own cache key and a custom fetch function, right?